src/main/java/net/consensys/wittgenstein/protocols/ENRGossiping.java

```diff
@@ -130,13 +130,31 @@ private void selectChangingNodes() {
     }
   }
 
+  private void addNewNode() {
+    ETHNode n = new ETHNode(network.rd, this.nb, generateCap());
+    network.addNode(n);
+    while (n.peers.size() < params.totalPeers) {
+      int peerId = network.rd.nextInt(network.allNodes.size());
+      if (!network.getNodeById(peerId).down)
+        network.createLink(n, network.getNodeById(peerId));
+    }
+    System.out.println("New node: " + n.nodeId + " has " + n.peers.size() + " peers");
+    n.start();
+  }
+
   @Override
   public void init() {
     for (int i = 0; i < params.NODES; i++) {
       network.addNode(new ETHNode(network.rd, this.nb, generateCap()));
     }
     network.setPeers();
 
+    selectChangingNodes();
```

Comment:
In selectChangingNodes we can select nodes that are down. This seems wrong.

Reply:
as we select nodes that change parameters prior to running (during the initial setup stage) all nodes are still up at that time.